### PR TITLE
Calendly & SavvyCal import

### PR DESCRIPTION
**Pull request by pumfleet**
This adds two buttons to the onboarding flow where you can choose which platform you'd like to import from, and runs two API queries to either API to get the user data and update the user with it, and the other API query gets your event types and creates them on our end.

Our goal is to make scheduling as open and accessible as possible and this helps customers make the right choice by not feeling locked in by other products.

### CODE REVIEW

File: pages/getting-started.tsx [hunk 1]
```diff
@@ -229,51 +232,130 @@ export default function Onboarding(props: inferSSRProps<typeof getServerSideProp
...
+              </Button>
+            </div>
+          )}
+          {selectedImport && (
+            <div>
+              <h2 className="text-2xl text-gray-900 font-cal">
+                Import from {selectedImport === "calendly" ? "Calendly" : "SavvyCal"}
+              </h2>
+              <p className="mb-2 text-sm text-gray-500">
+                You will need to generate an access token from the integrations page.
```
Comment (PeerRich) [inline pages/getting-started.tsx:267]: this needs `{t("string")}`

File: pages/getting-started.tsx [hunk 2]
```diff
@@ -229,51 +232,130 @@ export default function Onboarding(props: inferSSRProps<typeof getServerSideProp
...
-                <label htmlFor="timeZone" className="block text-sm font-medium text-gray-700">
-                  {t("timezone")}
+        <>
+          {selectedImport == "" && (
+            <div className="grid grid-cols-2 mb-4 gap-x-4">
+              <Button color="secondary" onClick={() => setSelectedImport("calendly")}>
+                Import from Calendly
+              </Button>
+              <Button color="secondary" onClick={() => setSelectedImport("savvycal")}>
+                Import from SavvyCal
```
Comment (PeerRich) [inline pages/getting-started.tsx:257]: missing string_key

File: pages/getting-started.tsx [hunk 3]
```diff
@@ -229,51 +232,130 @@ export default function Onboarding(props: inferSSRProps<typeof getServerSideProp
...
+                  }}
+                  type="text"
+                  name="token"
+                  id="token"
+                  placeholder={t("access_token")}
+                  required
+                  className="block w-full px-3 py-2 mt-1 border border-gray-300 rounded-sm shadow-sm focus:outline-none focus:ring-neutral-500 focus:border-neutral-500 sm:text-sm"
+                />
+                <Button type="submit" className="h-10 mt-1 ml-4">
+                  Import
```
Comment (PeerRich) [inline pages/getting-started.tsx:307]: missing string_key

**Comment by PeerRich** [comment]
very nice! can you add some description how to test this properly? maybe set up some test accounts too as I believe there is no way to retrigger the onboarding.

File: pages/api/import/savvycal.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  const authenticatedUser = await prisma.user.findFirst({
+    rejectOnNotFound: true,
+    where: {
+      id: session?.user.id,
+    },
+    select: {
+      id: true,
+    },
+  });
+  if (req.method == "POST") {
```
Comment (zomars) [inline pages/api/import/savvycal.ts:19]: Let's stick to strict equality

```suggestion
  if (req.method === "POST") {
```

File: pages/api/import/savvycal.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    const userResult = await fetch("https://api.savvycal.com/v1/me", {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: "Bearer " + req.body.token,
+      },
+    });
+
+    if (userResult.status == 200) {
```
Comment (zomars) [inline pages/api/import/savvycal.ts:28]:
```suggestion
    if (userResult.status === 200) {
```

**Review by zomars — COMMENTED** [review]
Code looks good, some screenshots would be nice tho :)

**Review by zomars — APPROVED** [review]

**Comment by edanfesi** [comment]
It looks good to me 🎉

**Review by edanfesi — APPROVED** [review]
We have to change to strict comparison but the rest is perfect

File: pages/getting-started.tsx [hunk 6]
```diff
@@ -229,51 +232,130 @@ export default function Onboarding(props: inferSSRProps<typeof getServerSideProp
...
+              </h2>
+              <p className="mb-2 text-sm text-gray-500">
+                You will need to generate an access token from the integrations page.
+              </p>
+              <form
+                className="flex"
+                onSubmit={formMethods.handleSubmit(async (values) => {
+                  setSubmitting(true);
+                  const response = await fetch(
+                    selectedImport === "calendly" ? "/api/import/calendly" : "/api/import/savvycal",
```
[the remark this answers is not in the document]
Comment (PeerRich) [reply pages/getting-started.tsx:272]: we should build a habit of screenshots and or loom videos @baileypumfleet

**Review by PeerRich — APPROVED** [review]

**Pull request merged**